blocks/editable/patterns.js

```diff
@@ -16,39 +21,22 @@ import { getBlockTypes } from '../api/registration';
  */
 const { setTimeout } = window;
 
-/**
- * Escapes characters for use in a Regular Expression.
- *
- * @param  {String} string Characters to escape
- *
- * @return {String}        Escaped characters
- */
-function escapeRegExp( string ) {
-	return string.replace( /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&' );
-}
-
 export default function( editor ) {
 	const getContent = this.getContent.bind( this );
 	const { onReplace } = this.props;
 
 	var VK = tinymce.util.VK;
 	var settings = editor.settings.wptextpattern || {};
 
-	const spacePatterns = getBlockTypes().reduce( ( acc, blockType ) => {
+	const patterns = getBlockTypes().reduce( ( acc, blockType ) => {
 		const transformsFrom = get( blockType, 'transforms.from', [] );
 		const transforms = transformsFrom.filter( ( { type } ) => type === 'pattern' );
 		return [ ...acc, ...transforms ];
 	}, [] );
 
-	var enterPatterns = settings.enter || [
-		// { start: '##', format: 'h2' },
-		// { start: '###', format: 'h3' },
-		// { start: '####', format: 'h4' },
-		// { start: '#####', format: 'h5' },
-		// { start: '######', format: 'h6' },
-		// { start: '>', format: 'blockquote' },
-		// { regExp: /^(-){3,}$/, element: 'hr' }
-	];
+	const [ enterPatterns, spacePatterns ] = partition( patterns, ( { regExp } ) =>
+		regExp.source.endsWith( '$' )
+	);
```

Comment:
nice!

although it's awkward to me splitting the function like this. was it too long for one line? I rarely split the arguments `( { regExp } )` from the function body via newline but that's personal style